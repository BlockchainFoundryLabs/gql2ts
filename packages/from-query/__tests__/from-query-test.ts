import Query from './data/query';
import Schema from './data/schema';
import FromQuery from '../src/index';

/**
 * original source: https://github.com/BinaryMuse/gql2ts-bug
 */
describe('additional fragments', () => {
  it ('generates a proper interface', () => {
    expect(
      FromQuery(Schema, Query)
    ).toMatchSnapshot();
  });
});
