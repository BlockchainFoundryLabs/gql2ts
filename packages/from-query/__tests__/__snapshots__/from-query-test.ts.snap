// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`additional fragments generates a proper interface 1`] = `
Array [
  Object {
    "additionalTypes": Array [
      "export interface SelectionOnOwner {
  login: string;
}
",
      "export interface IFragmentSpreadOnOrganization {
  __typename?: string;
}
",
      "export interface SelectionOnResource {
  __typename: string;
  id?: string;
  name?: string;
  owner?: SelectionOnOwner;
}
",
    ],
    "interface": "export interface MyTestQuery {
  resource:
    | SelectionOnResource &
        IFragmentPullRequest &
        Partial<IFragmentIssue> &
        Partial<IFragmentOrganization> &
        Partial<IFragmentSpreadOnOrganization>
    | null;
}
",
    "result": "export interface SelectionOnOwner {
  login: string;
}

export interface IFragmentSpreadOnOrganization {
  __typename?: string;
}

export interface SelectionOnResource {
  __typename: string;
  id?: string;
  name?: string;
  owner?: SelectionOnOwner;
}

export interface MyTestQuery {
  resource:
    | SelectionOnResource &
        IFragmentPullRequest &
        Partial<IFragmentIssue> &
        Partial<IFragmentOrganization> &
        Partial<IFragmentSpreadOnOrganization>
    | null;
}
",
    "variables": "",
  },
  Object {
    "additionalTypes": Array [
      "export interface SelectionOnAuthor {
  login: string;
}
",
    ],
    "interface": "export interface IFragmentIssue {
  id: string;
  title: string;
  author: SelectionOnAuthor | null;
}
",
    "result": "export interface SelectionOnAuthor {
  login: string;
}

export interface IFragmentIssue {
  id: string;
  title: string;
  author: SelectionOnAuthor | null;
}
",
    "variables": "",
  },
  Object {
    "additionalTypes": Array [],
    "interface": "export interface IFragmentOrganization {
  orgName: string | null;
}
",
    "result": "export interface IFragmentOrganization {
  orgName: string | null;
}
",
    "variables": "",
  },
  Object {
    "additionalTypes": Array [],
    "interface": "export interface IFragmentPullRequest {
  id?: string;
  title?: string;
  author?: SelectionOnAuthor | null;
}
",
    "result": "export interface IFragmentPullRequest {
  id?: string;
  title?: string;
  author?: SelectionOnAuthor | null;
}
",
    "variables": "",
  },
]
`;
